### PR TITLE
[DOCS] Add 'Change DS mappings and settings' tutorial

### PR DESCRIPTION
**Pull request by jrodewig**
Adds a tutorial for updating the mappings and
index settings of a data stream's backing indices.

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/data-streams/change-mappings-and-settings.asciidoc [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+backing index, which is auto-generated when the stream is created.
+
+Before creating a data stream, we recommend you carefully consider which
+mappings and settings to include in this template. However, if you later need to
+change the mappings or settings of a data stream's backing indices, you have a
+couple options:
+
+* To apply changes to future backing indices, simply update the composable
+template associated with the data stream. Mapping and setting changes will be
+automatically applied to any backing indices created after the update.
```
Comment (danhermann) [inline docs/reference/data-streams/change-mappings-and-settings.asciidoc:19]: Just a heads-up: this is currently accurate though we're considering applying mappings at the level of the data stream instead of for each backing index.

File: docs/reference/data-streams/change-mappings-and-settings.asciidoc [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+`logs` data stream. This creates a new write index with mappings and index
+settings from the recently updated `logs_data_stream` template.
+
+[source,console]
+----
+POST /logs/_rollover/
+{
+  "conditions": {
+    "max_docs":   "1"
+  }
```
Comment (danhermann) [inline docs/reference/data-streams/change-mappings-and-settings.asciidoc:136]: This is correct, but if the intent is to unconditionally roll over the data stream, you can omit the `conditions` block altogether.

**Review by danhermann — APPROVED** [review]
LGTM. Left some informational comments below but nothing that requires changes.

File: docs/reference/data-streams/change-mappings-and-settings.asciidoc [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+====
+
+. Use the <<indices-create-data-stream,create data stream API>> to manually
+create the new data stream. The name of the data stream must match the name or
+wildcard pattern defined in the new template's `index_patterns` property.
++
+IMPORTANT: If you do not want to mix new and old data, do not use an indexing
+request to create the data stream. Mixing old and new data is safe, but the
+backing index containing this data needs to be retained until you are ready to
+delete the new data.
```
Comment (andreidan) [inline docs/reference/data-streams/change-mappings-and-settings.asciidoc:259]: I don't understand this. Is this still related to reindexing into the new data stream? Is the new data the data that'll be sent directly to the new data stream (having not been ported over from the old data stream)? If that's the case where would we delete it from?

File: docs/reference/data-streams/change-mappings-and-settings.asciidoc [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+// TEST[continued]
+====
+
+. If you do not want to mix new and old data in your new data stream, pause the
+indexing of new documents.
+
+. If you use {ilm-init} to <<getting-started-index-lifecycle-management,automate
+rollover>>, reduce the {ilm-init} poll interval to ensure that the current write
+index doesn’t grow too large while waiting for the rollover check. By default,
+{ilm-init} checks rollover conditions every 10 minutes.
```
Comment (andreidan) [inline docs/reference/data-streams/change-mappings-and-settings.asciidoc:282]: this ties into my previous comment. I think we need to provide more details about new/old data and what we're trying to achieve here

File: docs/reference/data-streams/change-mappings-and-settings.asciidoc [hunk 5]
```diff
@@ -0,0 +1,520 @@
...
+////
+====
+
+. Reindex your data to the new data stream using an `op_type` of `create`.
++
+If you want to partition the data in the order in which it was originally
+indexed, you can run separate reindex requests. These reindex requests can use
+individual backing indices as the source. You can use the
+<<indices-get-data-stream,get data stream API>> to retrieve a list of backing
+indices.
```
Comment (andreidan) [inline docs/reference/data-streams/change-mappings-and-settings.asciidoc:324]: Do we want to promote this concept? If we talked about ILM managing the data stream (which presumably has a rollover policy) do we want to talk about backing indices? (which are an implementation details for data streams)

**Review by andreidan — APPROVED** [review]
Thanks for documenting this James. This looks great!

I left a few minor comments and concerns (I'm a bit confused by a couple of paragraphs)

File: docs/reference/data-streams/change-mappings-and-settings.asciidoc (hunk 5 above)
Comment (danhermann) [reply docs/reference/data-streams/change-mappings-and-settings.asciidoc:324]: I think we have to strike a careful balance between completely hiding the fact that data streams include multiple backing indices (which I think would be unhelpful) and always talking about operating on individual backing indices (which would eliminate much of the value of data streams).

In this particular case, I can see scenarios where using individual backing indices as reindex sources would be useful. E.g., when you have a data stream with a large number of backing indices and you want to be able to be able to easily resume the reindex process if it fails at any point during the operation. Perhaps in the future we could make the reindex operation handle resuming operations on data streams without needing to reference individual backing indices, but that would certainly be an enhancement for a future phase.

File: docs/reference/data-streams/change-mappings-and-settings.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/data-streams/change-mappings-and-settings.asciidoc:19]: Thanks for noting that. I'll leave as-is for now, but I think we could easily remove the mapping references later on.

File: docs/reference/data-streams/change-mappings-and-settings.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/data-streams/change-mappings-and-settings.asciidoc]: Thanks for catching this. Updated with e015f4f0e9b.

File: docs/reference/data-streams/change-mappings-and-settings.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/data-streams/change-mappings-and-settings.asciidoc]: Thanks for raising this @andreidan. I've revised this and the next relevant section.

To answer your question, I was envisioning a scenario in which a user creates the new data stream by indexing new documents to it. Because the user will later reindex data from an existing data stream into this new stream, it could result in one or more backing indices that contain mixed data.

This could interfere with data retention if the user deletes older backing indices. A user could prematurely delete an index they assume contains only old data but actually contain new data as well. This could result in unintended data loss.

The alternative is to keep such a backing indices until its newest data expires. This could increase storage costs and potentially violate their data retention policies.

Comment (andreidan) [reply docs/reference/data-streams/change-mappings-and-settings.asciidoc]: Thanks for the explanation and for revising this @jrodewig. It makes perfect sense now and I think it's great this sits in its own section now

**Comment by jrodewig** [comment]
### Backport commits
master 0fc2bd8 
7.x c548a87
